kolibri/core/content/utils/transfer.py

```diff
@@ -149,17 +153,26 @@ def __init__(self, *args, **kwargs):
             # initialize a fresh requests session, if one wasn't provided
             self.session = requests.Session()
 
+        # Record the size of content that has been transferred
+        self.transferred_size = 0
+
         super(FileDownload, self).__init__(*args, **kwargs)
 
     def start(self):
-        # If a file download was stopped by Internet connection error,
-        # then open the temp file again.
-        if self.started:
-            self.dest_file_obj = open(self.dest_tmp, "wb")
-
         # initiate the download, check for status errors, and calculate download size
-        self.response = self.session.get(self.source, stream=True, timeout=self.timeout)
-        self.response.raise_for_status()
+        try:
+            self.response = self.session.get(
+                self.source, stream=True, timeout=self.timeout
+            )
+            self.response.raise_for_status()
+        except Exception as e:
+            retry = retry_import(e)
+            if not retry:
+                raise
+            # Catch exceptions to check if we should resume file downloading
+            self.resume()
+            if self.cancel_check():
+                self._kill_gracefully()
```

Comment:
Should this cancel check also happen in the case of successful `get`? (i.e. be de-indented) -- or is there a reason not to?

Reply:
@jamalex yes, the cancel check would happen as well when `get` is successful, but in a bit later time here: https://github.com/learningequality/kolibri/pull/6470/files#diff-4d578307b2a82fe54231716093334b43R200. If a user hits `cancel` button after the `get` is successful, it will not cancel the process right away, but will do so before we start to actually import the content, since the `next()` function is called right after `start()`. Do you have any suggestions if there is a better way to handle this case? Thank you!

Reply:
That sounds good!

Reply:
thank you!